### ISSUE TITLE
ATSAMD51 hardware SPI not working

### ISSUE THREAD

**Issue by NuclearPhoenixx** [adafruit/Adafruit_FRAM_SPI#11]
Hi, I've been recently working with the ATSAMD51 and come to test it with some FRAM. However, it seems like the hardware spi mode has some issues.

- Arduino board:  ItsyBitsy M4 (SAMD51)

- Arduino IDE version: 1.8.11 (latest)

- Reproduce the problem: 

Load the example info sketch. Using "hardware spi" the samd seems to freeze and does not open any serial connections. If you use the "software spi" with manual pins it seems to work. At least I'm getting serial connections again. Compilation is successfull in both cases.

**Comment by ladyada** [comment]
try changing this speed to 1MHz?
https://github.com/adafruit/Adafruit_FRAM_SPI/blob/master/Adafruit_FRAM_SPI.cpp#L109

**Comment by NuclearPhoenixx** [comment]
Doesn't change anything, same problem :/

**Comment by RobertKieboom** [comment]
I am having exactly the same problem on the feather M0 (ATSAMD21). Software SPI works, hardware SPI doesn't. It doesn't work on a genuino zero either. Other SPI devices (like you micro SD breakout) do work on both. I checked on an Arduino Uno and that works too. The ATSAMD devices hang in the begin (I confirmed that with debugging on the zero with PlatformIO). It makes no difference if I use PlatformIO or the Arduino IDE. I have the latest version of the library installed.
